dll/win32/browseui/shellbars/CBandSiteMenu.cpp

```diff
@@ -97,8 +97,17 @@ HRESULT CBandSiteMenu::_CreateMenuPart()
 
         /* Get the band name */
         StringCchPrintfW(wRegKey, MAX_PATH, L"CLSID\\%s", wszBandGUID);
-        dwDataSize = MAX_PATH;
-        SHGetValue(HKEY_CLASSES_ROOT, wRegKey, NULL, NULL, wszBandName, &dwDataSize);
+        HKEY hKey;
+        if (RegOpenKeyExW(HKEY_CLASSES_ROOT, wRegKey, 0, KEY_READ, &hKey) == ERROR_SUCCESS)
+        {
+            hr = SHLoadRegUIStringW(hKey, L"MenuTextPUI", wszBandName, _countof(wszBandName));
+            if (FAILED_UNEXPECTEDLY(hr))
+            {
+                SHLoadRegUIStringW(hKey, NULL, wszBandName, _countof(wszBandName));
+            }
+            RegCloseKey(hKey);
+        }
+        wszBandName[_countof(wszBandName) - 1] = UNICODE_NULL; /* Avoid buffer overrun */
```

Comment:
This kind of thing should have been ensured by the SHLoadRegUIStringW call,
but not also that if your RegOpenKeyExW failed to begin with, the wszBandName buffer could have garbage stored in all bytes until its last (i.e. byte index `_countof(wszBandName) - 1`) one.

Reply:
Removed.
Fixed by using `SHGetValue`.